src/malli/core.cljc

```diff
@@ -2469,22 +2469,24 @@
    (swap! -function-schemas* assoc-in [key ns name] (merge data {:schema (f ?schema), :ns ns, :name name}))))
 
 #?(:clj
-   (defmacro => [name value]
+   (defmacro => [given-sym value]
      (let [cljs-resolve (when (:ns &env) (ns-resolve 'cljs.analyzer.api 'resolve))
            cljs-resolve-symbols (fn [env d]
                                   (walk/postwalk (fn [x] (cond->> x (symbol? x) (or (:name (cljs-resolve env x)))))
                                                  d))
-           name' `'~(symbol (str name))
-           ns' `'~(symbol (str *ns*))
-           sym `'~(symbol (str *ns*) (str name))
+           name-str (name given-sym)
+           ns-str (str (or (not-empty (namespace given-sym)) *ns*))
+           name' `'~(symbol name-str)
+           ns' `'~(symbol ns-str)
+           sym `'~(symbol ns-str name-str)
            value' (cond->> value (:ns &env) (cljs-resolve-symbols &env))]
        ;; in cljs we need to register the schema in clojure (the cljs compiler)
        ;; so it is visible in the (function-schemas :cljs) map at macroexpansion time.
        (if (:ns &env)
          (do
-           (-register-function-schema! (symbol (str *ns*)) name value' (meta name) :cljs identity)
-           `(do (-register-function-schema! ~ns' ~name' ~value' ~(meta name) :cljs identity) ~sym))
-         `(do (-register-function-schema! ~ns' ~name' ~value' ~(meta name)) ~sym)))))
+           (-register-function-schema! (symbol ns-str) (symbol name-str) value' (meta given-sym) :cljs identity)
+           `(do (-register-function-schema! ~ns' ~name' ~value' ~(meta given-sym) :cljs identity) ~sym))
```

Comment:
line 2488 should have the same value as line 2487 (`ns-str` and `name-str`) then I think this should work in cljs.

Reply:
If I change this to match 2487, other stuff breaks. To be clear, they're fundamentally doing the same thing, but `(symbol ns-str)` is executed during macroexpansion/eval, and `~ns'` is macroexpanded to the symbol value of `ns-str` (roughly `(list 'quote ~(symbol ns-str))`) and executed at run-time.

Reply:
To fix the test, I had to change the namespace from `clojure.core` to `cljs.core`.

Reply:
ahh ok thanks for clarifying - I see what's happening now!